shell/shared/admin.js

```diff
@@ -214,10 +313,22 @@ if (Meteor.isServer) {
     }
   };
 
+  var updateLoginStyleToRedirect = function (serviceName) {
+    var configurations = Package["service-configuration"].ServiceConfiguration.configurations;
+    var config = configurations.findOne({service: serviceName});
+
+    if (config) {
```

Comment:
Maybe also check if loginStyle is already "redirect", in which case you can skip setting it?